mkdocs_static_i18n/suffix.py

```diff
@@ -53,7 +54,10 @@ def reconfigure_file(
     else:
         file.dest_path = file_dest_path
     file.abs_dest_path = os.path.normpath(os.path.join(config.site_dir, file.dest_path))
-    file.name = Path(file_dest_path).stem
```

Comment:
This one impacts the "unlocalization" of files. Since we're removing language suffix from them, it's important to align their name as well

Before your patch :
```
INFO     -  reconfigure File(src_uri='topic2/1.1.filename.fr.html', dest_uri='de/topic2/1.1.filename.html', name='1.1.filename', url='de/topic2/1.1.filename.html') from
```

After your patch :
```
INFO     -  reconfigure File(src_uri='topic2/1.1.filename.fr.html', dest_uri='de/topic2/1.1.filename.html', name='1.1.filename.fr', url='de/topic2/1.1.filename.html') from
```

Reply:
OK, I shall see what I can do. Before my changes `file.name` was `index` for almost every file. This lead the `navigation.index` Material feature to break. Maybe something else influenced it 🤔 

BTW I've noticed that it's a debug log and you displayed it as INFO. Is there a way to limit the log output to the plugin / exclude other plugins?
`log.debug(f"reconfigure {file} from locale {file_locale}")`

Reply:
> This lead the navigation.index Material feature to break.

Well then there must be something wrong in my own testing. My understanding of this feature is that it allows to NOT show a useless indentation of the "index" section in the sidebar. With the initial file.name cleanup, this is what happens indeed but if I remove it (your proposed code), it does.

> BTW I've noticed that it's a debug log and you displayed it as INFO. Is there a way to limit the log output to the plugin / exclude other plugins?

No, so I just shamelessly locally switched it to INFO :blush:

Reply:
What do you mean, by useless section. If the feature is turned on it merges the `index` page with the section "header". We have a lot of unmerged and merged sections on our page.
https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/nav-item.html#L65-L66
It uses the `nav_item.is_index`, which looks at the `file.name` being `index` xD

Basically this is incorrect:
```py
reconfigure 
    File(
    src_uri='topic1/named_file.en.md', 
    dest_uri='de/topic1/named_file/index.html', 
    name='index', 
    url='de/topic1/named_file/') 
from locale en
```
because the name is taken from the destination, which in most cases is `index.html`.
The best I can do right now is `file.name = PurePath(file.name).stem`, but then this breaks another link
```py
reconfigure 
    File(
    src_uri='topic2/1.1.filename.html', 
    dest_uri='de/topic2/1.1.filename.html', 
    name='1.1', 
    url='de/topic2/1.1.filename.html') 
from locale en
```
also the locale is set to `en` while in actuality it doesn't have the `en` suffix. iirc before the refactor the plugin used `None` here 🤔

Reply:
> What do you mean, by useless section. If the feature is turned on it merges the index page with the section "header".

I mean exactly that "merge = remove useless", your term is better.

Reply:
> also the locale is set to en while in actuality it doesn't have the en suffix. iirc before the refactor the plugin used None here

Yes, but it was confusing since we're actually interested by the locale we're building

Reply:
> We have a lot of unmerged and merged sections on our page.

How can I "see" this and reproduce? Could you propose to add some files to the test folder structure to demonstrate this?

Reply:
> How can I "see" this and reproduce? Could you propose to add some files to the test folder structure to demonstrate this?

For some reason I get slightly delayed notifications, and the PR comments don't reload ad hoc... anyways
Imo there is no need for additional files. If a file is called `index.md` inside of a Section, then it should be merge with the header, if it's not called `index.md` then it should not be merged with the Section.

https://github.com/Gothic-Modding-Community/gmc/tree/dev/docs/zengin/scripts/classes <- none of the files are merged with the header and it behaves like on your current mkdocs page, the Section header isn't a URL

https://github.com/Gothic-Modding-Community/gmc/tree/dev/docs/zengin/worlds <- index merges with the header the rest are beneath

https://gothic-modding-community.github.io/gmc/zengin/ interactive example 😄 

Also here comes the kicker, if a page merges with the Section, the title is still taken from the Section and not the page. This seems to actually work, if I seen it correctly when the `file.name` was broken.

Reply:
OK I need to think about how to test that, reproduce properly via a test and then fix it for good

Reply:
If you can access the files list in the test. Compare the source path with the `file.name`. Only README and index should have the `index` `file.name`. Good luck :v: 
I'll try to think about how to fix configuration across builds tomorrow, and I really want to try to modify the JavaScript. Both MkDocs and Material have a static implementation of it since years. Material could introduce some changes in the coming future, but still it shouldn't be something needing a lot of maintenance.
This way only `i18n` should require preservation.

Reply:
Also after sleeping on the issue at hand I figured. Why is `file.name` being something else than `index` suddenly a problem. If the "unlocalization" requires the `file.name` to be `Path(dest_path).stem` use the `.stem` value there instead of the `file.name` 😵